apps/src/lib/script-editor/LessonCard.jsx

```diff
@@ -235,6 +246,7 @@ export class UnconnectedLessonCard extends Component {
           <OrderControls
             type={ControlTypes.Lesson}
             position={lesson.position}
+            parentPosition={lessonGroupPosition}
```

Comment:
for another time... it's a bit weird for the OrderControls to need to know the parentPosition. however, I can't blame you because it OrderControls doesn't really need to know its own position either, and you are following the same pattern. Ideally LessonCard would define:
```js
  move = direction => {
    this.props.moveLesson(this.props.parentPosition, this.props.position, direction)
  }
  ...
  render() {
    <OrderControls move = this.props.moveLesson/>
  }
  ...
  connect(
    dispatch => ({
      moveLesson(position, parentPosition, direction) {
        dispatch(moveLesson(parentPosition, position, direction));
      }
    })
  )
```
this would let you skip all of the type-checking logic in in the `connect` method on OrderControls too (explicit type checking is a good sign that things may not be encapsulated properly).

Reply:
Working on this in a follow up PR: https://github.com/code-dot-org/code-dot-org/pull/34893 (it grew bigger than I wanted to tackle in this PR)